### PR TITLE
link ndl2 github page in documentation

### PR DESCRIPTION
**Pull request by derNarr**
ndl2 is now on github, this PR links it in the documentation

### CODE REVIEW

**Review by dekuenstle — DISMISSED** [review]
Great, thank you @derNarr ! A big push for open science :-)

File: README.rst [hunk 1]
```diff
@@ -82,8 +82,17 @@ Currently, it is maintained by `Konstantin Sering <https://github.com/derNarr>`_
 and `Marc Weitz <https://github.com/trybnetic>`_.
 
 
+Funding
+=======
+*pyndl* supported by the Alexander von Humboldt Professorship awarded to R.
```
Comment (Trybnetic) [inline README.rst:87]: has been supported or is supported?

File: README.rst [hunk 2]
```diff
@@ -82,8 +82,17 @@ Currently, it is maintained by `Konstantin Sering <https://github.com/derNarr>`_
...
+Harald Baayen, the ERC advanced Grant (no. 742545), and by the University of
+Tübingen.
+
 Acknowledgments
 ===============
-This research was supported by an ERC advanced Grant (no. 742545) and by the
-Alexander von Humboldt Professorship awarded to R. H. Baayen and by the
-University of Tübingen.
+This package is build as a python replacement for the R `ndl2 package
```
Comment (Trybnetic) [inline README.rst:93]: as a Python replacement

File: README.rst [hunk 3]
```diff
@@ -82,8 +82,17 @@ Currently, it is maintained by `Konstantin Sering <https://github.com/derNarr>`_
...
+<https://github.com/quantling/ndl2>`_. Some
+ideas on how to build the API and how to efficiently run the Rescorla Wagner
+iterative learning on large text corpora are inspired by the way the ndl2
+package solves this problems.
```
Comment (Trybnetic) [inline README.rst:97]: these problems

File: docs/source/benchmark.rst [hunk 4]
```diff
@@ -89,8 +89,9 @@ the iterative learner of `ndl2`. ``R -f run_ndl.R`` runs the `ndl2` on the same
 while `ndl` only runs on the smaller files because it is orders of magnitude slower.
 
 
-`ndl` can be installed from CRAN by running ``install.packages('ndl')`` in R, while `ndl2` is just
-available upon request (contact `Konstantin Sering <https://uni-tuebingen.de/fakultaeten/philosophische-fakultaet/fachbereiche/neuphilologie/seminar-fuer-sprachwissenschaft/arbeitsbereiche/quantitative-linguistik/mitarbeiter/'>`_, install with ``R CMD INSTALL ndl2.tar.gz``).
+`ndl` can be installed from CRAN by running ``install.packages('ndl')`` in R,
+while `ndl2` needs to be downloaded from Github first (`ndl2
+<https://github.com/quantling/ndl2>`_, install with ``R CMD INSTALL ndl2.tar.gz``).
```
Comment (Trybnetic) [inline docs/source/benchmark.rst:94]: Isn't there an option to install R packages from Github directly?

File: docs/source/credits.rst [hunk 5]
```diff
@@ -58,14 +58,15 @@ If you are using BibTex you may want to use this example BibTex entry::
 
 Funding
 -------
-*pyndl* was partially funded by the Humboldt grant, the ERC advanced grant (no.
-742545) and by the University of Tübingen.
-
+*pyndl* supported by the Alexander von Humboldt Professorship awarded to R.
```
Comment (Trybnetic) [inline docs/source/credits.rst:61]: same issue as above. Most likely ''has been supported by''

File: docs/source/credits.rst [hunk 6]
```diff
@@ -58,14 +58,15 @@ If you are using BibTex you may want to use this example BibTex entry::
...
 Acknowledgements
 ----------------
-This package is build as a python replacement for the R ndl2 package. Some
+This package is build as a python replacement for the R `ndl2 package
+<https://github.com/quantling/ndl2>`_. Some
 ideas on how to build the API and how to efficiently run the Rescorla Wagner
 iterative learning on large text corpora are inspired by the way the ndl2
-package solves this problems. The ndl2 package will be published to github soon
-and a reference will be placed here.
+package solves this problems.
```
Comment (Trybnetic) [inline docs/source/credits.rst:71]: these problems

**Review by Trybnetic — CHANGES_REQUESTED** [review]
Thanks a lot @derNarr. I made some comments regarding language.

File: docs/source/benchmark.rst (hunk 4 above)
Comment (derNarr) [reply docs/source/benchmark.rst:94]: Yes, in principle it is possible to run ``github_install('quantling/ndl2')`` but this only succeed, if all dependencies are installed (as far as I understand). If you download the package from github manually you are exposed to the instructions that dependencies exist.

Comment (Trybnetic) [reply docs/source/benchmark.rst:94]: I see. I just remembered that ''back in the days'', I used `devtools::install_github()` a few times and it seemed to work, but I do not remember anymore whether you had to install the dependencies manually or not. Probably, its worth trying out.

File: docs/source/credits.rst [hunk 7]
```diff
@@ -58,15 +58,14 @@ If you are using BibTex you may want to use this example BibTex entry::
 
 Funding
 -------
-*pyndl* supported by the Alexander von Humboldt Professorship awarded to R.
-Harald Baayen, the ERC advanced Grant (no. 742545), and by the University of
-Tübingen.
+*pyndl* has been supported by the Alexander von Humboldt Professorship awarded
+to R.  Harald Baayen, the ERC advanced Grant WIDE (no. 742545), and the
```
Comment (Trybnetic) [inline docs/source/credits.rst:62]: Being very pedantic: There is a whitespace to much after the R.

I am not sure whether `sphinx` removes this while building. If it does, I am fine with it

**Comment by derNarr** [comment]
I had to do:

```bash
sudo apt-get install r-cran-devtools
```

and then

```R
library(devtools)
devtools::install_github('quantling/ndl2')
```

This gives me this error message:

```
Downloading GitHub repo quantling/ndl2@HEAD
✔  checking for file ‘/tmp/RtmpFbrjVv/remotes1f67548ee37a1/quantling-ndl2-5f26ad0/DESCRIPTION’ ...
─  preparing ‘ndl2’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  looking to see if a ‘data/datalist’ file should be added
─  building ‘ndl2_0.1.1.tar.gz’
   
Installing package into ‘/home/tino/R/x86_64-pc-linux-gnu-library/4.2’
(as ‘lib’ is unspecified)
* installing *source* package ‘ndl2’ ...
** using staged installation
** libs
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c common/helper.cpp -o common/helper.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c common/serialization.cpp -o common/serialization.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c common/Exception.cpp -o common/Exception.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c common/view_events.cpp -o common/view_events.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c danks.cpp -o danks.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c learning/rescorla_wagner.cpp -o learning/rescorla_wagner.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c learning/file_management.cpp -o learning/file_management.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c learning/util.cpp -o learning/util.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c learning/matrix_io.cpp -o learning/matrix_io.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c learning/events_manager.cpp -o learning/events_manager.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c learning/r_wrapper.cpp -o learning/r_wrapper.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c preprocessing/ndlpreproc.cpp -o preprocessing/ndlpreproc.o
preprocessing/ndlpreproc.cpp: In function ‘std::string getFormattedTime(tm*)’:
preprocessing/ndlpreproc.cpp:1539:31: warning: ‘%02d’ directive output may be truncated writing between 2 and 11 bytes into a region of size between 8 and 15 [-Wformat-truncation=]
 1539 |     snprintf(buffer, 20, "%4d-%02d-%02d %02d:%02d:%02d", 1900+ptm->tm_year,
      |                               ^~~~
preprocessing/ndlpreproc.cpp:1539:26: note: directive argument in the range [-2147483647, 2147483647]
 1539 |     snprintf(buffer, 20, "%4d-%02d-%02d %02d:%02d:%02d", 1900+ptm->tm_year,
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:867,
                 from /usr/include/c++/10/cstdio:42,
                 from preprocessing/ndlpreproc.cpp:10:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:67:35: note: ‘__builtin___snprintf_chk’ output between 20 and 72 bytes into a destination of size 20
   67 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   68 |        __bos (__s), __fmt, __va_arg_pack ());
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c preprocessing/ndlOutBuffer.cpp -o preprocessing/ndlOutBuffer.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c view_events_wrapper.cpp -o view_events_wrapper.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I'/home/tino/R/x86_64-pc-linux-gnu-library/4.2/Rcpp/include'   -fopenmp -DUSING_RCPP=1 -Wall -fpic  -g -O2 -ffile-prefix-map=/home/jranke/git/r-backports/bullseye/r-base-4.2.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c preprocessing_wrapper.cpp -o preprocessing_wrapper.o
make: *** No rule to make target 'RcppExports.o', needed by 'ndl2.so'.  Stop.
ERROR: compilation failed for package ‘ndl2’
* removing ‘/home/tino/R/x86_64-pc-linux-gnu-library/4.2/ndl2’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpFbrjVv/file1f675655196fa/ndl2_0.1.1.tar.gz’ had non-zero exit status
```

**Comment by Trybnetic** [comment]
I see. Thanks for trying out. Maybe, we find a simpler way later, but this seems anyway more relevant for the ndl2 repository. See the double whitspace issue, but I am also fine without resolving that.

**Review by Trybnetic — APPROVED** [review]

**Comment by derNarr** [comment]
After I installed the dependencies as described in https://github.com/quantling/ndl2/releases/tag/v0.1.1 with

```bash
sudo apt-get install libboost-filesystem-dev libboost-system-dev libboost-regex-dev libicu-dev
```
Running `devtools::install_github('quantling/ndl2')` still does not work. But it works to install the downloaded `ndl2.tar.gz` with:

```bash
R CMD INSTALL ndl2.tar.gz 
```

**Comment by Trybnetic** [comment]
And should we bump to v1.0.2?

**Pull request merged**